lstchain/io/io.py

```diff
@@ -361,9 +361,12 @@ def write_array_info(event, output_filename):
         if len(ids) > 0:  # only write if there is a telescope with this camera
             tel_id = list(ids)[0]
             camera = sub.tel[tel_id].camera
+            camera_name = str(camera)
+            if camera_name is 'UNKNOWN':
```

Comment:
with this line, what you are doing is _not_ to write any of the camera information into the file. I guess this is ok because when the camera name is UNKNOWN, this information does not exist, right?

Reply:
exactly, that happens when ctapipe does not recognize the telescope.

Reply:
since SimTel files don't contain camera names, there is just a lookup table in ctapipe, which returns UNKNOWN if no matches are found.  Perhaps you shold open an issue for that - we could for example expose an API to allow you to add your own in the case they are not defined.  In fact, right now, you can just modify `ctapipe.instrument.guess.TELESCOPE_NAMES` inside your code, before you try to load the SimTel file. Just add an entry for your new telescope.  The GuessingKey is just a tuple of `(num_pixels, focal_length)`

Reply:
To be clearer, just add the line:
```py3
from ctapipe.instrument.guess import TELESCOPE_NAMES, GuessingKey, GuessingResult

TELESCOPE_NAMES[GuessingKey(2048, 2.28)] =  GuessingResult("LST", "Custom", "MyCamera", 2)
```

(filling in the correct number of pix and focal length), and then you will get a telescope named "LST_Custom_MyCamera".

Reply:
Thank you for the trick Karl, I had to modify ctapipe code to get the same result. 
Defining the telescope works well but on the LST + Magic production the 2 magic telescopes are slightly different but share the same focal length and the same number of pixel and so trigger the same bug.
The latest commits I pushed handle exactly that.

Reply:
> Defining the telescope works well but on the LST + Magic production the 2 magic telescopes are slightly different but share the same focal length and the same number of pixel and so trigger the same bug.
> The latest commits I pushed handle exactly that.

@yrenier I think camera-geometry wise the two telescopes should be identical - they have slightly different characteristicas in other respects (is that what you meant?), but that should not affect the camera geometry selection.

Reply:
The purpose of the TelescopeDescription is really to describe "classes" of telescope, so if the MAGIC telescopes have similar mirrors, cameras, and focal length, I would still call them both "MAGIC", and then you get 2 tel-ids, (1 and 2) that let you distiguish between the two telescopes.   For the combined LST-Magic array, we might need to define what tels get which id number (e..g [1,2] = MAGIC, and 3=LST?), but other than that I think the reconstruciton should still work fine.

Reply:
what I noticed was that the iteration 
`for telescope_type in sub.telescope_types:`
was iterating on each of the the MAGIC telescope. 
If I add the MAGIC telescope as suggested by @kosack both get the same name.
If i don't add them they get the same name: UNKNONW
Either way the  write_array_info() function crash when trying to write the 2nd camera with  camera.to_table().write() as the /instrument/telescope/camera/{camera_name} key already exists.
I can provide a small sim_tearrayl file so you can reproduce the problem if you want.